scripts/zones/Port_Jeuno/npcs/Sagheera.lua

```diff
@@ -332,15 +362,53 @@ entity.onTrigger = function(player, npc)
     then
         player:startEvent(313)
 
-    -- DEFAULT DIALOG (menu)
+    -- Prevent interaction until player has progressed through COP enough
+    elseif player:getCurrentMission(xi.mission.log_id.COP) < xi.mission.id.cop.GARDEN_OF_ANTIQUITY then
+        player:showText(npc, ID.text.SAGHEERA_NO_LIMBUS_ACESS)
+
+        -- DEFAULT DIALOG (menu)
     else
         -- event parameters
+
+        local storedABCs = player:getCurrency("ancient_beastcoin")
+        local playerSagheera = player:getCharVar("Sagheera")
```

Comment:
is it essential this be a charvar and not a localvar, are we needing this to persist beyond the event? sort of skimming atm from my dayjob

Reply:
I do believe it needs to be a charvar as it handles the first time you talk to her she has special dialog to sell you a cosmo cleanse and then she gives you a long explanation in other places that I'm fairly certain only happen once ever and not every time you zone.

Reply:
To avoid a forever charvar, we could set a value when completing the requisite mission, and then reset to 0

Reply:
Ok, so I'll set this value with all important bits when the player finishes the requisite mission and then remove the bits as the player performs the necessary interactions. Sound right?

Reply:
Yep, sounds good, that way we don't clutter the db after the player has progressed far enough

Reply:
does she do multiple of those one offs? it may be justified as a forever var if so

Reply:
> Ok, so I'll set this value with all important bits when the player finishes the requisite mission and then remove the bits as the player performs the necessary interactions. Sound right?

yeah :+1: